source/ios/AdaptiveCards/ADCIOSVisualizer/ADCIOSVisualizerUITests/ADCIOSVisualizerUITests.m

```diff
@@ -32,9 +28,100 @@ - (void)tearDown {
     [super tearDown];
 }
 
-- (void)testExample {
-    // Use recording to get started writing UI tests.
-    // Use XCTAssert and related functions to verify your tests produce the correct results.
+- (void)verifyChoiceSetInput:(NSDictionary<NSString *, NSString *> *) expectedValue application:(XCUIApplication *)app {
+    NSData *expectedData = [NSJSONSerialization dataWithJSONObject:expectedValue options:NSJSONWritingPrettyPrinted error:nil];
+    NSString *expectedString = [[NSString alloc] initWithData:expectedData encoding:NSUTF8StringEncoding];
+    XCUIElement *queryResult = app/*@START_MENU_TOKEN@*/.scrollViews.otherElements[@"ACR Root View"]/*[[".scrollViews.otherElements[@\"ACR Root View\"]",".otherElements[@\"ACR Root View\"]"],[[[-1,1],[-1,0]]],[1]]@END_MENU_TOKEN@*/.staticTexts[expectedString];
+    XCTAssertTrue([queryResult.label isEqualToString:expectedString]);
```

Comment:
What are all the weird comments in this code for? They make it pretty hard to read.

Reply:
they are auto generated by XCode UI recording feature.

---
In reply to: [211801393](https://github.com/Microsoft/AdaptiveCards/pull/1799#discussion_r211801393) [](ancestors = 211801393)